core/src/main/java/com/sequenceiq/cloudbreak/domain/InstanceMetaData.java

```diff
@@ -41,7 +41,13 @@
                 query = "SELECT i FROM InstanceMetaData i "
                         + "WHERE i.instanceGroup.stack.id= :stackId "
                         + "AND i.instanceGroup.groupName= :groupName "
-                        + "AND (i.instanceStatus= 'DECOMMISSIONED' OR i.instanceStatus= 'UNREGISTERED')")
+                        + "AND (i.instanceStatus= 'DECOMMISSIONED' OR i.instanceStatus= 'UNREGISTERED')"),
+        @NamedQuery(
+                name = "InstanceMetaData.findByPrivateAndPublicAddress",
+                query = "SELECT i FROM InstanceMetaData i "
+                        + "WHERE i.instanceGroup.stack.id= :stackId "
+                        + "AND i.privateIp= :privateAddress "
+                        + "AND i.publicIp= :publicAddress ")
```

Comment:
Why is it needed to find by both private and public address? It's not possible that two instances in the same stack have the same private address, so I guess the public address is redundant.